doc/api/http2.md

```diff
@@ -0,0 +1,727 @@
+
+* `stream` {Http2Stream}
+* `options` {Object}
+  * `exclusive` {boolean} (TODO: fill in detail)
+  * `parent` {number} (TODO: fill in detail)
+  * `weight` {number} (TODO: fill in detail)
+
+Updates the priority for the given `Http2Stream` instance. If `options.silent`
+is `false`, causes a new `PRIORITY` frame to be sent to the connected HTTP/2
+peer.
+
+#### http2session.submitSettings(settings)
+
+* `settings` {[Settings Object][]}
+* Returns {undefined}
+
+Updates the current local settings for this `Http2Session` and sends a new
+`SETTINGS` frame to the connected HTTP/2 peer.
+
+#### http2session.type
+
+* Value: {number}
+
+The `http2session.type` will be equal to
+`http2.constants.NGHTTP2_SESSION_SERVER` if this `Http2Session` instance is a
+server, and `http2.constants.NGHTTP2_SESSION_CLIENT` if the instance is a
+client.
+
+### Class: Http2Stream
+
+* Extends: {Duplex}
+
+Each instance of the `Http2Stream` class represents a bidirectional HTTP/2
+communications stream over an `Http2Session` instance. Any single `Http2Session`
+may have up to 2<sup>31</sup>-1 `Http2Stream` instances over its lifetime.
+
+User code will not construct `Http2Stream` instances directly. Rather, these
+are created, managed, and provided to user code through the `Http2Session`
+instance. On the server, `Http2Stream` instances are created either in response
+to an incoming HTTP request (and handed off to user code via the `'stream'`
+event), or in response to a call to the `http2stream.pushStream()` method.
+On the client, `Http2Stream` instances are created and returned when either the
+`http2session.request()` method is called, or in response to an incoming
+`'push'` event.
+
+#### Event: 'aborted'
+
+The `'aborted'` event is emitted whenever a `Http2Stream` instance is
+abnormally aborted in mid-communication.
+
+#### Event: 'error'
+
+(TODO: fill in detail)
+
+#### Event: 'fetchTrailers'
+
+The `'fetchTrailers`' event is emitted by the `Http2Stream` immediately after
+queuing the last chunk of payload data to be sent. The listener callback is
+passed a single object (with a `null` prototype) that the listener may used
+to specify the trailing header fields to send to the peer.
+
+```js
+stream.on('fetchTrailers', (trailers) => {
+  trailers['ABC'] = 'some value to send';
+});
+```
+
+#### Event: 'headers'
+
+The `'headers'` event is emitted when a block of headers has been received
+on the `Http2Stream`, and the block does not correspond with an HTTP request,
+response or push request. The listener callback is passed the [Headers Object][]
+and flags associated with the headers.
+
+```js
+stream.on('headers', (headers, flags) => {
+  // ...
+});
+```
+
+#### Event: 'push'
+
+The `'push'` event is emitted when response headers for a Server Push stream
+are received. The listener callback is passed the [Headers Object][] and flags
+associated with the headers.
+
+```js
+stream.on('push', (headers, flags) => {
+  // ...
+});
+```
```

Comment:
agreed.... but let's add that separately once the push stuff is a bit more settled.